src/lib/Configuration/Provider/AlloyEditor.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformRichText\Configuration\Provider;
+
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use EzSystems\EzPlatformRichText\Configuration\UI\Mapper\OnlineEditorConfigMapper;
+use EzSystems\EzPlatformRichText\SPI\Configuration\Provider;
+use EzSystems\EzPlatformRichTextBundle\DependencyInjection\Configuration\Parser\FieldType\RichText;
+
+/**
+ * AlloyEditor configuration provider.
+ *
+ * @internal For internal use by RichText package
+ */
+final class AlloyEditor implements Provider
+{
+    /** @var array */
+    private $alloyEditorConfiguration;
+
+    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
+    private $configResolver;
+
+    /** @var \EzSystems\EzPlatformRichText\Configuration\UI\Mapper\OnlineEditorConfigMapper */
+    private $onlineEditorConfigMapper;
+
+    /**
+     * @param array $alloyEditorConfiguration
+     * @param \eZ\Publish\Core\MVC\ConfigResolverInterface $configResolver
+     * @param \EzSystems\EzPlatformRichText\Configuration\UI\Mapper\OnlineEditorConfigMapper $onlineEditorConfigMapper
+     */
+    public function __construct(
+        array $alloyEditorConfiguration,
+        ConfigResolverInterface $configResolver,
+        OnlineEditorConfigMapper $onlineEditorConfigMapper
+    ) {
+        $this->alloyEditorConfiguration = $alloyEditorConfiguration;
+        $this->configResolver = $configResolver;
+        $this->onlineEditorConfigMapper = $onlineEditorConfigMapper;
+    }
+
+    public function getName(): string
+    {
+        return 'alloyEditor';
+    }
+
+    /**
+     * @return array AlloyEditor config
+     */
+    public function getConfiguration(): array
+    {
+        return [
+            'extraPlugins' => $this->getExtraPlugins(),
+            'extraButtons' => $this->getExtraButtons(),
+            'classes' => $this->getCssClasses(),
+            'attributes' => $this->getDataAttributes(),
+        ];
+    }
+
+    /**
+     * @return array Custom plugins
+     */
+    private function getExtraPlugins(): array
+    {
+        return $this->alloyEditorConfiguration['extra_plugins'] ?? [];
+    }
+
+    /**
+     * @deprecated 3.0.0 The alternative and more flexible solution will be introduced.
+     * @deprecated 3.0.0 So you will need to update Online Editor Extra Buttons as part of eZ Platform 3.x upgrade.
+     *
+     * @return array Custom buttons
+     */
+    private function getExtraButtons(): array
+    {
+        @trigger_error(
+            '"ezrichtext.alloy_editor.extra_buttons" is deprecated since v2.5.1. There will be new and more flexible solution to manage buttons in Online Editor in 3.0.0',
```

Comment:
can you split this into lines to make it more readable?

Reply:
Actually, I've decided to modify behavior here a bit as well. Till now it reported that warning on every load of AdminUI, even if user didn't set that configuration. So I've changed it to check if it's used in the first place. See 6acd3d112. This will reduce size of deprecation log.